feat: Update quickstart guides (#328)
strimzi/strimzi.github.io: quickstarts/minikube.md

Minikube provides a local Kubernetes, designed to make it easy to learn and develop for Kubernetes.
The Kubernetes cluster is started inside a virtual machine, running a Linux kernel and a Docker daemon.

# Installing the dependencies

> This quickstart assumes that you have the latest version of the `minikube` binary, which you can get [here](https://minikube.sigs.k8s.io/docs/start/).

Reviewer: ```suggestion
This quickstart assumes that you have the latest version of the `minikube` binary, which you can get from the [minikube website](https://minikube.sigs.k8s.io/docs/start/).
```

Minikube requires a container or virtual machine manager.
The Minikube documentation includes a list of suggested options in the [getting started guide](https://minikube.sigs.k8s.io/docs/start/).

> You'll also need the `kubectl` binary, which you can get by following the instructions [here](https://kubernetes.io/docs/tasks/tools/).

Reviewer: ```suggestion
You'll also need the `kubectl` binary, which you can get by following the [`kubectl` installation instructions](https://kubernetes.io/docs/tasks/tools/) from the Kubernetes website.
```

Once you have all the binaries installed, make sure everything works:

```shell
# Validate minikube
minikube version

# Validate kubectl
kubectl version
```

> # Starting Kubernetes cluster

Reviewer: ```suggestion
# Starting the Kubernetes cluster
```

> This will start a local development cluster of [Minikube](https://minikube.sigs.k8s.io/docs/start/) which runs in a container or virtual machine manager.

Reviewer: ```suggestion
Start a local development cluster of [Minikube](https://minikube.sigs.k8s.io/docs/start/) that runs in a container or virtual machine manager.
```

```shell
minikube start --memory=4096 # 2GB default memory isn't always enough
```

# Applying Strimzi installation file

Before deploying Strimzi Kafka operator, let's first create our `kafka` namespace:

```shell
kubectl create namespace kafka
```

Next we apply the Strimzi install files, including `ClusterRoles`, `ClusterRoleBindings` and some **Custom Resource Definitions** (`CRDs`). The CRDs define the schemas used for declarative management of the Kafka cluster, Kafka topics and users.

```shell
kubectl create -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka
```

This will be familiar if you've installed Strimzi before.
Note how we set all the `namespace` references in downloaded .yaml file to `kafka`. By default they are set to `myproject`.
But we want them all to be `kafka` because we decided to install the operator into the `kafka` namespace, which we achieve by specifying `-n kafka` when running `kubectl create` ensuring that all the definitions and the configurations are installed in `kafka` namespace rather than the `default` namespace.

> If there is a mismatch between namespaces, then the Strimzi Cluster Operator will not have the necessary permissions to perform its operations.

Reviewer: Same comments as for the Docker Desktop

Follow the deployment of the Strimzi Kafka operator:
```shell
kubectl get pod -n kafka --watch
```

You can also follow the operator's log:
```shell
kubectl logs deployment/strimzi-cluster-operator -n kafka -f
```

# Provision the Apache Kafka cluster

> Then we create a new Kafka custom resource, which will give us a small persistent Apache Kafka Cluster with one node for each - Apache Zookeeper and Apache Kafka:

Reviewer: ```suggestion
Configure a new Kafka custom resource to create a small persistent Apache Kafka Cluster with one node for Apache Zookeeper and Apache Kafka:
```

Minikube provides a local Kubernetes, designed to make it easy to learn and develop for Kubernetes.

> The Kubernetes cluster is started inside a virtual machine, running a Linux kernel and a Docker daemon.

Reviewer 2: This is not that easy anymore. By default (if installed) it will use Docker driver.

Author: Do you think the following is more representative of the current state of minikube:
```
The Kubernetes cluster is started either inside a virtual machine, a container or on bare-metal, depending on the minikube driver you choose.
```

Reviewer 2: Yeah, I think that works.

# Installing the dependencies

This quickstart assumes that you have the latest version of the `minikube` binary, which you can get from the [minikube website](https://minikube.sigs.k8s.io/docs/start/).

Minikube requires a container or virtual machine manager.
The Minikube documentation includes a list of suggested options in the [getting started guide](https://minikube.sigs.k8s.io/docs/start/).

You'll also need the `kubectl` binary, which you can get by following the [`kubectl` installation instructions](https://kubernetes.io/docs/tasks/tools/) from the Kubernetes website.

Once you have all the binaries installed, make sure everything works:

```shell
# Validate minikube
minikube version

# Validate kubectl
kubectl version

> ```

Author: Both commands list the current version that's installed, so in some ways the output is fairly self explanatory. I don't think it's obvious how to best explain what the output should be since the version I have for each won't match the version the reader is using. Happy to hear any suggestions though

# Starting the Kubernetes cluster

Start a local development cluster of [Minikube](https://minikube.sigs.k8s.io/docs/start/) that runs in a container or virtual machine manager.

```shell
minikube start --memory=4096 # 2GB default memory isn't always enough
```

# Deploy Strimzi using installation files

Before deploying the Strimzi cluster operator, create a namespace called `kafka`:

```shell
kubectl create namespace kafka
```

> Apply the Strimzi install files, including `ClusterRoles`, `ClusterRoleBindings` and some **Custom Resource Definitions** (`CRDs`). The CRDs define the schemas used for declarative management of the Kafka cluster, Kafka topics and users.

Author: I've added this suggestion

```shell
kubectl create -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka
```

The YAML files for `ClusterRoles` and `ClusterRoleBindings` downloaded from strimzi.io contain a default namespace of `myproject`.
The query parameter `namespace=kafka` updates these files to use `kafka` instead.
By specifying `-n kafka` when running `kubectl create`, the definitions and configurations without a namespace reference are also installed in the `kafka` namespace.
If there is a mismatch between namespaces, then the Strimzi cluster operator will not have the necessary permissions to perform its operations.

Follow the deployment of the Strimzi cluster operator:
```shell
kubectl get pod -n kafka --watch
```

You can also follow the operator's log:
```shell
kubectl logs deployment/strimzi-cluster-operator -n kafka -f
```

# Configure an Apache Kafka cluster

> Configure a new Kafka custom resource to create a small persistent Apache Kafka Cluster with one node for Apache Zookeeper and Apache Kafka:

Reviewer: Yep. Create works for me here. :+1:

# Send and receive messages

With the cluster running, run a simple producer to send messages to a Kafka topic (the topic is automatically created):

```shell
kubectl -n kafka run kafka-producer -ti --image=quay.io/strimzi/kafka:{{site.data.releases.operator[0].version}}-kafka-{{site.data.releases.operator[0].defaultKafkaVersion}} --rm=true --restart=Never -- bin/kafka-console-producer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic my-topic
```

And to receive them in a different terminal, run:

```shell
kubectl -n kafka run kafka-consumer -ti --image=quay.io/strimzi/kafka:{{site.data.releases.operator[0].version}}-kafka-{{site.data.releases.operator[0].defaultKafkaVersion}} --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic my-topic --from-beginning
```

Enjoy your Apache Kafka cluster, running on Minikube!

# Deleting your Apache Kafka cluster

When you are finished with your Apache Kafka cluster, you can delete it by running:

```shell
kubectl -n kafka delete kafka my-cluster
```

> This will remove the Apache Kafka cluster but leave the Strimzi cluster operator running so that it can respond to new Kafka custom resources.

Reviewer 2: Should we here and in the others also include the operator deletion?
```
kubectl delete -f 'https://strimzi.io/install/latest?namespace=kafka' -n kafka
```

> 

Reviewer 3: Should we have instructions for deleting the operator installation too, or deleting the minikube cluster?

Enjoy your Apache Kafka cluster, running on Minikube!

# Deleting your Apache Kafka cluster

> When you are finished with your Apache Kafka cluster you can delete it by running:

Reviewer: ```suggestion
When you are finished with your Apache Kafka cluster, you can delete it by running:
```

```shell
kubectl -n kafka delete kafka my-cluster
```

> This will remove the Apache Kafka cluster but leave the Strimzi Cluster Operator running so that it can respond to new Kafka CRs.

Reviewer: ```suggestion
This will remove the Apache Kafka cluster, but leave the Strimzi Cluster Operator running so that it can respond to new Kafka custom resources.
```
